test/Microsoft.ML.TestFramework/DataPipe/TestDataPipeBase.cs

```diff
@@ -906,11 +906,11 @@ protected Func<bool> GetColumnComparer(IRow r1, IRow r2, int col, ColumnType typ
                 case DataKind.Bool:
                     return GetComparerOne<DvBool>(r1, r2, col, (x, y) => x.Equals(y));
                 case DataKind.TimeSpan:
-                    return GetComparerOne<DvTimeSpan>(r1, r2, col, (x, y) => x.Equals(y));
+                    return GetComparerOne<TimeSpan>(r1, r2, col, (x, y) => x.Ticks == y.Ticks);
```

Comment:
>TimeSpan [](start = 42, length = 8)

As near as I can see, both `TimeSpan` and `DateTime` are directly comparable, that might be better.